### PR TITLE
INDY-953: Fix protocol version upgrade

### PR DESCRIPTION
**Pull request by ashcherbakov**
fix issues with upgrade:

   - do not send protocol version with NODE_UPGRADE txns (since some nodes in the pool may not yet understand it)
  -  do not reschedule the same upgrade
  -  do not re-send NODE_UPGRADE after each view change

### CODE REVIEW

File: indy_node/test/upgrade/test_node_upgrade_in_progress_no_protocol_version.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from indy_common.constants import IN_PROGRESS
+from indy_node.test.helper import TestNode
+from indy_node.test.upgrade.helper import check_node_sent_acknowledges_upgrade
+
+whitelist = ['unable to send message']
+
+
+class TestNodeNoProtocolVersion(TestNode):
```
Comment (spivachuk) [inline indy_node/test/upgrade/test_node_upgrade_in_progress_no_protocol_version.py]: It makes sense to move this class to `indy_node.test.upgrade.helper` module in order not to duplicate it in tests.

File: indy_node/test/upgrade/test_node_upgrade_no_protocol_version.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import indy_node
+from indy_common.constants import COMPLETE
+from indy_node.test.helper import TestNode
+from indy_node.test.upgrade.helper import populate_log_with_upgrade_events, \
+    check_node_sent_acknowledges_upgrade
+
+whitelist = ['unable to send message']
+
+version = indy_node.__metadata__.__version__
```
Comment (andkononykhin) [inline indy_node/test/upgrade/test_node_upgrade_no_protocol_version.py]: the same as above

File: indy_node/server/upgrader.py [hunk 3]
```diff
@@ -113,19 +112,32 @@ def __repr__(self):
     def service(self):
         return self._serviceActions()
 
-    def check_upgrade_succeeded(self):
+    def process_upgrade_log_for_first_run(self):
+        # whether upgrade was started before the Node restarted,
+        # that is whether Upgrade Log contains STARTED event
+        self._upgrade_started = self._is_upgrade_started()
+        if self._upgrade_started:
+            # append SUCCESS or FAIL to the Upgrade Lof
```
Comment (spivachuk) [inline indy_node/server/upgrader.py]: Please correct: Lof -> Log.

File: indy_node/test/upgrade/helper.py [hunk 4]
```diff
@@ -96,10 +96,20 @@ def __init__(self, backup_dir, backup_target, transform=lambda tool: None):
...
         self.p = multiprocessing.Process(target=self.tool.start)
         self.p.start()
+        logger.debug("NCTProcess was started with pid: {}". format(self.p.pid))
 
     def stop(self):
-        self.p.terminate()
+        logger.debug("Send stop to NCTProcess with pid: {}". format(self.p.pid))
         self.tool.server.close()
+        self.p.terminate()
+        # check that process with NetControlTool.start function really stop.
```
Comment (spivachuk) [inline indy_node/test/upgrade/helper.py]: Please correct: NetControlTool -> NodeControlTool.

File: indy_node/test/upgrade/test_node_upgrade_in_progress_no_protocol_version.py (hunk 1 above)
Comment (ashcherbakov) [reply indy_node/test/upgrade/test_node_upgrade_in_progress_no_protocol_version.py]: Agree. Let's do it in a separate PR.

File: indy_node/test/upgrade/helper.py (hunk 4 above)
Comment (ashcherbakov) [reply indy_node/test/upgrade/helper.py]: Agree. Let's do it in a separate PR.

File: indy_node/server/upgrader.py [hunk 5]
```diff
@@ -285,60 +298,70 @@ def handleUpgradeTxn(self, txn) -> None:
...
+
+            if self.scheduledUpgrade:
+                logger.info(
+                    "Node '{}' cancels previous upgrade and schedules a new one to {}".format(
+                        self.nodeName, version))
+                self._cancelScheduledUpgrade(justification)
+
+            self._scheduleUpgrade(
+                version, when, failTimeout, upgrade_id)
+            return
```
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply indy_node/server/upgrader.py]: Agree. Let's do it in a separate PR.

File: indy_node/test/upgrade/test_node_upgrade.py [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
+from indy_node.server.upgrade_log import UpgradeLog
+from indy_common.constants import COMPLETE
+from indy_node.test.upgrade.helper import populate_log_with_upgrade_events, \
+    check_node_sent_acknowledges_upgrade, check_node_do_not_sent_acknowledges_upgrade, \
+    emulate_restart_pool_for_upgrade, emulate_view_change_pool_for_upgrade
 
 whitelist = ['unable to send message']
 # TODO: Implement a client in node
 
+version = indy_node.__metadata__.__version__
```
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply indy_node/test/upgrade/test_node_upgrade.py]: Agree. Let's do it in a separate PR.

**Pull request merged**